doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,107 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
    .. note::
 
       In the case of nRF9160 DK, the reset button must be pressed to restart the device and to start the application.
+
+.. _gs_debugging:
+
+Debugging an application
+************************
+
+To debug an application, set up the debug session as described in `Debugging an application`_ in the |nRFVSC| documentation.
+
+If you use a multi-core SoC, for example from the nRF53 Series, and you only wish to debug the application core firmware, a single debug session is sufficient.
+To debug the firmware running on the network core, you need to set up two separate debug sessions: one for the network core and one for the application core.
+When debugging the network core, the application core debug session runs in the background and you can debug both cores if needed.
+
+Complete the following steps to start debugging the network core:
+
+1. Set up sessions for the application core and network core as mentioned in `Debugging an application`_.
+#. Select the appropriate CPU for debugging in each session, corresponding to the application core and the network core of your SoC, respectively.
+#. Once both sessions are established, execute the code on the application core.
+
+   The startup code releases the ``NETWORK.FORCEOFF`` signal to start the network core and allocates the necessary GPIO pins for it.
+#. Start code execution on the network core in the other debug session.
+
+If you want to reset the network core while debugging, make sure to first reset the application core and execute the code.
+
+Debugging secure and non-secure firmware
+========================================
+
+When using an ``_ns`` build target, by default you can only debug firmware in the non-secure part of the application core firmware.
+
+To debug firmware running in the secure part, you need to build Trusted Firmware-M with debug symbols enabled and load the symbols during the debugging session.
+To build Trusted Firmware-M with debug symbols, set the :kconfig:option:`CONFIG_TFM_CMAKE_BUILD_TYPE_RELWITHDEBINFO` Kconfig option.
+To load the Trusted Firmware-M debug symbols in the GDB tool, run the following command and answer ``y`` if prompted by the tool:
+
+.. code-block:: console
+
+   (gdb) add-symbol-file build/tfm/bin/tfm_s.elf
+
+For more information about the GDB tool, see `Debug tools`_.
+
+Debug configuration
+===================
+
+The following Kconfig options are automatically set to ``y`` by the |VSC| debugger in the |nRFVSC| when you enable the debug configuration before building the application:
+
+* :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` - This option limits the optimizations made by the compiler to only those that do not impact debugging.
+* :kconfig:option:`CONFIG_DEBUG_THREAD_INFO` - This option adds additional information to the thread object, so that the debugger can discover the threads.
+  This will work for any debugger.
+
+If you are programming from command line, you can set these options to ``y`` manually.
```

Comment:
> programming from command line.

when you say programming from command line, I assume you refer to `west flash`, `nrfjprog` or similar.
How is it possible to set build options when programming ?

And what will happen if I program from command line, and afterwards launch the debugger in VSCode ?
When developing an app, those are different steps:
- compiling and linking (building)
- programming (write image to flash)
   - programming may rebuild an app if there are code changes, but this depends on user settings.
- debugging (attach debugger to running program)
  - launching a debug session may re-program the flash
  - attaching a debugger may reset the SoC

from the current text, it is not clear what each step actually does, and hence it becomes must harder to know what happens when.